great_expectations/experimental/datasources/postgres_datasource.py

```diff
@@ -116,19 +117,18 @@ class TableAsset(DataAsset):
     table_name: str
     column_splitter: Optional[ColumnSplitter] = None
     name: str
-    _order_by: List[BatchSorter] = pydantic.PrivateAttr()
-
-    def __init__(self, **kwargs):
-        # I `pop("order_by", None) or []` instead of `pop("order_by", [])` because if someone
-        # passes in `order_by=None`, I want this variable to be `[]` and not `None`.
-        # `pop("order_by", [])` will return None since the order_by key exists in this case.
-        order_by = kwargs.pop("order_by", None) or []
-        self._order_by = _batch_sorter_from_list(order_by)
-        super().__init__(**kwargs)
-
-    @property
-    def order_by(self) -> List[BatchSorter]:
-        return self._order_by
+    order_by: List[BatchSorter] = Field(default_factory=list)
```

Comment:
Mutable default values for pydantic fields are actually ok.
https://stackoverflow.com/a/63808835/6304433

We could do this as below...
```python
order_by: List[BatchSorter] = []
```
but unless they know `pydantic` well, this will upset most pythonistas so using the `Field` `default_factory` is more explicit.

Reply:
Appreciate the context here 🙇🏽 

I say let's stay away from mutable defaults just because they're such a footgun with vanilla Python

Reply:
Just to clarify for other readers.
mutable defaults ARE NOT a footgun when used with pydantic.

Both of the examples below are immune from the usual problems associated with [mutable default values](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) in python.
We are only choosing the second example because it's more expressive/explicit.
```python
order_by_a: List[BatchSorter] = []
order_by_b: List[BatchSorter] = Field(default_factory=list)
```